drivers/watchdog/wdt_sdw_apb.c

```diff
@@ -0,0 +1,346 @@
+	wdt_sdw_user_tops[12] = DT_PROP_HAS_IDX(WDOG_0, utops, 12);
+	wdt_sdw_user_tops[13] = DT_PROP_HAS_IDX(WDOG_0, utops, 13);
+	wdt_sdw_user_tops[14] = DT_PROP_HAS_IDX(WDOG_0, utops, 14);
+	wdt_sdw_user_tops[15] = DT_PROP_HAS_IDX(WDOG_0, utops, 15);
+
+	return wdt_sdw_user_tops;
+}
+
+
+static int wdog_sdw_init_timeouts(const struct device *dev,
+				  unsigned long clk_rate)
+{
+	volatile struct wdog_sdw_apb *wdog = WDOG_STRUCT;
+	uint32_t data;
+	const uint32_t *tops;
+
+	/*
+	 * use fixed top if WDT_USE_FIX_TOP set 1
+	 * otherwise use custom tops
+	 */
+	data = wdog->comp_param_1;
+	if (data & WDOG_COMP_PARAMS_1_USE_FIX_TOP) {
+		tops = wdt_sdw_fix_tops;
+	} else {
+		if (DT_NODE_HAS_PROP(WDOG_0, utops)) {
+			tops = get_user_tops();
+		} else {
+			tops = wdt_sdw_fix_tops;
+		}
+	}
+
+	/* calculatei timeouts using top values */
+	wdog_sdw_handle_tops(dev, clk_rate, tops);
+	if (!timeouts[WDT_SDW_NUM_TOPS - 1].sec) {
+		printk("No any valid TOP detected\n");
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int wdog_sdw_apb_setup(const struct device *dev, uint8_t options)
+{
+	volatile struct wdog_sdw_apb *wdog = WDOG_STRUCT;
+
+	ARG_UNUSED(dev);
+
+	/* set reset pulse length */
+	if ((options >= WDOG_RPL_PCLK_CYCLES_2) &&
+	    (options <= WDOG_RPL_PCLK_CYCLES_256)) {
+		wdog->ctrl |= (options << WDOG_CTRL_RPL_SHIFT);
+	}
+
+	/* Enable the watchdog counter with INTEN bit */
+	wdog->ctrl |= WDOG_CONTROL_REG_WDT_EN_MASK;
+
+	return 0;
+}
+
+static int wdog_sdw_apb_disable(const struct device *dev)
+{
+	volatile struct wdog_sdw_apb *wdog = WDOG_STRUCT;
+
+	ARG_UNUSED(dev);
+
+	/* Stop the watchdog counter */
+	wdog->ctrl = ~(WDOG_CONTROL_REG_WDT_EN_MASK);
+
+	return 0;
+}
+
+static int wdog_sdw_apb_install_timeout(const struct device *dev,
+					const struct wdt_timeout_cfg *config)
+{
+	int ret;
+
+	ARG_UNUSED(dev);
+
+	ret = wdog_sdw_init_timeouts(dev,
+				     DT_INST_PROP_BY_PHANDLE(0,
+							     clocks,
+							     clock_frequency));
+	if (ret < 0)
+		return ret;
+	ret = wdog_sdw_set_timeout(dev, config->window.max);
+	if (ret < 0)
+		return ret;
+
+	wdog_sdw_update_mode(dev, config->flags);
+
+	/* Configure only the callback */
+	user_cb = config->callback;
+
+	return 0;
+}
+
+static int wdog_sdw_apb_feed(const struct device *dev, int channel_id)
+{
+	volatile struct wdog_sdw_apb *wdog = WDOG_STRUCT;
+
+	ARG_UNUSED(dev);
+	ARG_UNUSED(channel_id);
+
+	/* Reload */
+	wdog->crr = WDOG_COUNTER_RESTART_KICK_VALUE;
+
+	return 0;
+}
+
+static const struct wdt_driver_api wdog_sdw_apb_api = {
+	.setup = wdog_sdw_apb_setup,
+	.disable = wdog_sdw_apb_disable,
+	.install_timeout = wdog_sdw_apb_install_timeout,
+	.feed = wdog_sdw_apb_feed,
+};
+
+static int wdog_sdw_apb_init(const struct device *dev)
+{
+	/* Configure the wdog mode */
+	wdog_sdw_update_mode(dev, WDOG_RMOD_RESET);
+
+#ifdef CONFIG_WDOG_SDW_APB_START_AT_BOOT
+	wdog_sdw_apb_setup(dev, 0);
+#endif
+	return 0;
+}
+
+DEVICE_DT_INST_DEFINE(0,
```

Comment:
Please use DT_INST_FOREACH_STATUS_OKAY so that the user can still prevent device allocation using `status = "disabled";`.